### PR TITLE
Ensure user is instructor before sensitive / modification actions

### PR DESCRIPTION
**Pull request by adambray**

### CODE REVIEW

**Comment by adambray** [comment]
Fixes #22

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -31,6 +31,13 @@ def authorize_user!
     return false
   end
 
+  def authorize_instructor!
+    return true if is_an_instructor? || has_api_token?
```
Comment (jshawl) [inline app/controllers/application_controller.rb]: students query their info with an access_token as well. see https://github.com/wdidc/status/blob/7a8063f095b7bdebf654e7ad730278334215c0c8/js/app.js#L73 does this affect that?

Comment (adambray) [inline app/controllers/application_controller.rb]: That uses `submissions#index`, which is exempted from the `authorize_instructor!` call.

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class SubmissionsController < ApplicationController
+  before_filter :authorize_instructor!, except: [:index]
```
Comment (adambray) [inline app/controllers/submissions_controller.rb]: Here is where the exemption takes place.

Comment (jshawl) [inline app/controllers/submissions_controller.rb]: :+1:

**Comment by jshawl** [comment]
:+1: good to merge!

**Pull request merged**